File: Java_ChildrenCompetition/src/main/java/repository/IParticipantRepository.java
package repository;

import model.Participant;

//public interface IParticipantRepository<ID, T extends HasId<ID>> extends IRepository<ID, T>{
//    T findByUsername(String username);
//    Iterable<T> findAllParticipantsForTest(int id);
//}

public interface IParticipantRepository extends IRepository<Integer, Participant>{
    Participant findByUsername(String username);
    Iterable<Participant> findAllParticipantsForTest(int id);
}

File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/repository/IParticipantRepository.cs
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public interface IParticipantRepository <ID, T> : IRepository<ID, T> where T : HasId<ID>
    {
        T findByUsername(string username);
        IEnumerable<T> findAllParticipantsForTest(int id);
    }
}